include/tdyts.h

```diff
@@ -0,0 +1,25 @@
+#if !defined(TIMESTEPPER_H)
+#define TIMESTEPPER_H
+
+#include <petsc.h>
+
+typedef struct Timestepper *Timestepper;
+
+struct Timestepper {
+  SNES snes;
```

Comment:
This structure would eventually also include `TS ts;`. So, we need an integer that identifies if `SNESSet/Solve` or `TSSet/Solve` would be called. So, how about adding an integer that identifies the type of timestepper (e.g. `PetscInt solver_type`).